api_app/analyzers_manager/observable_analyzers/ip2location.py

```diff
@@ -0,0 +1,45 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class Ip2location(classes.ObservableAnalyzer):
+    base_url: str = "https://api.ip2location.io/"
+    _api_key_name: str
+
+    def get_response(self, payload):
+        return requests.get(self.base_url, params=payload)
+
+    def run(self):
+        try:
+            payload = {"ip": self.observable_name}
+
+            # Add key to payload if the API version is "keyed"
+            if self.ip2location_api_version == "keyed":
```

Comment:
you need to define this parameter as an attribute of the class. Please explain the difference between the keyless and the keyed option

Reply:
done please take a look